apps/examples/testcase/le_tc/kernel/tc_libc_timer.c

```diff
@@ -415,6 +415,108 @@ static void tc_libc_timer_strftime(void)
+*                  structure as per the format specifiers.
+* API's covered   :strptime
+* Preconditions   :none
+* Postconditions  :none
+* @return         :void
+*/
+static void tc_libc_timer_strptime(void)
+{
+	struct tm sp_tm;
+	struct tm sf_tm;
+	char *ret;
+
+	sf_tm.tm_sec  = 13;
+	sf_tm.tm_min  = 42;
+	sf_tm.tm_hour = 12;
+	sf_tm.tm_mday = 5;
+	sf_tm.tm_mon  = 5;
+	sf_tm.tm_year = 2018 - YEAR_BASE;
+
+	memset(&sp_tm, 0, sizeof(struct tm));
+	ret = strptime("2018-06-05 12:42:13 PM", "%Y-%m-%Od %k:%M:%S %p", &sp_tm);
+	TC_ASSERT_NEQ("strptime", ret, NULL);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_year, sf_tm.tm_year);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_mon, sf_tm.tm_mon);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_mday, sf_tm.tm_mday);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_hour, sf_tm.tm_hour);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_min, sf_tm.tm_min);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_sec, sf_tm.tm_sec);
+
+	memset(&sp_tm, 0, sizeof(struct tm));
+	ret = strptime("22 June 10 AM 18", "%U %B %l %p %Ey", &sp_tm);
+	TC_ASSERT_NEQ("strptime", ret, NULL);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_year, sf_tm.tm_year);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_mon, sf_tm.tm_mon);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_hour, sf_tm.tm_hour - 2);
+
+	memset(&sp_tm, 0, sizeof(struct tm));
+	ret = strptime("2018", "%C", &sp_tm);
+	TC_ASSERT_NEQ("strptime", ret, NULL);
+
+	memset(&sp_tm, 0, sizeof(struct tm));
+	ret = strptime("06/05/18 12:42:13", "%D %X", &sp_tm);
+	TC_ASSERT_NEQ("strptime", ret, NULL);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_year, sf_tm.tm_year);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_mon, sf_tm.tm_mon);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_mday, sf_tm.tm_mday);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_hour, sf_tm.tm_hour);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_min, sf_tm.tm_min);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_sec, sf_tm.tm_sec);
+
+	memset(&sp_tm, 0, sizeof(struct tm));
+	ret = strptime("06/05/18 12:42", "%x %R", &sp_tm);
+	TC_ASSERT_NEQ("strptime", ret, NULL);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_year, sf_tm.tm_year);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_mon, sf_tm.tm_mon);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_mday, sf_tm.tm_mday);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_hour, sf_tm.tm_hour);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_min, sf_tm.tm_min);
+
+	memset(&sp_tm, 0, sizeof(struct tm));
+	ret = strptime("12:42:13", "%T", &sp_tm);
+	TC_ASSERT_NEQ("strptime", ret, NULL);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_hour, sf_tm.tm_hour);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_min, sf_tm.tm_min);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_sec, sf_tm.tm_sec);
+
+	memset(&sp_tm, 0, sizeof(struct tm));
+	ret = strptime("12:42:13 PM", "%r", &sp_tm);
+	TC_ASSERT_NEQ("strptime", ret, NULL);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_hour, sf_tm.tm_hour);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_min, sf_tm.tm_min);
+	TC_ASSERT_EQ("strptime", sp_tm.tm_sec, sf_tm.tm_sec);
+
+#if defined(CONFIG_LIBC_LOCALTIME) || defined(CONFIG_TIME_EXTENDED)
+	sf_tm.tm_wday = 2; // Tuesday (0-6: day of the week, week starts on Sunday)
```

Comment:
can you put the comment in c style and above this line of code , instead of  keeping it on same line ?